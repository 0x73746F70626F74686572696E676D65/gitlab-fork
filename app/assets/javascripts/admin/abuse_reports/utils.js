import { FILTERED_SEARCH_TOKEN_CATEGORY } from './constants';

export const buildFilteredSearchCategoryToken = (categories) => {
  const options = categories.map((c) => ({ value: c, title: c }));
  return { ...FILTERED_SEARCH_TOKEN_CATEGORY, options };
};
