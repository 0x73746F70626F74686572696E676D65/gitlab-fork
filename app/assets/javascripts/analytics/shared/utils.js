import dateFormat from 'dateformat';
import { dateFormats } from './constants';

export const filterBySearchTerm = (data = [], searchTerm = '', filterByKey = 'name') => {
  if (!searchTerm?.length) return data;
  return data.filter((item) => item[filterByKey].toLowerCase().includes(searchTerm.toLowerCase()));
};

export const toYmd = (date) => dateFormat(date, dateFormats.isoDate);
