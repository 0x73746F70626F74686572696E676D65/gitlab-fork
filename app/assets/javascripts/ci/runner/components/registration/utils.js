/* eslint-disable @gitlab/require-i18n-strings */
import {
  DEFAULT_PLATFORM,
  LINUX_PLATFORM,
  MACOS_PLATFORM,
  WINDOWS_PLATFORM,
  DOWNLOAD_LOCATIONS,
} from '../../constants';
import linuxInstall from './scripts/linux/install.sh?raw';
import osxInstall from './scripts/osx/install.sh?raw';
import windowsInstall from './scripts/windows/install.ps1?raw';

const OS = {
  [LINUX_PLATFORM]: {
    commandPrompt: '$',
    executable: 'gitlab-runner',
  },
  [MACOS_PLATFORM]: {
    commandPrompt: '$',
    executable: 'gitlab-runner',
  },
  [WINDOWS_PLATFORM]: {
    commandPrompt: '>',
    executable: '.\\gitlab-runner.exe',
  },
};

export const commandPrompt = ({ platform }) => {
  return (OS[platform] || OS[DEFAULT_PLATFORM]).commandPrompt;
};

export const executable = ({ platform }) => {
  return (OS[platform] || OS[DEFAULT_PLATFORM]).executable;
};

export const registerCommand = ({ platform, url = gon.gitlab_url, registrationToken }) => {
  return [
    `${executable({ platform })} register`,
    `  --url ${url}`,
    ...(registrationToken ? [`  --registration-token ${registrationToken}`] : []),
  ];
};

export const runCommand = ({ platform }) => {
  return `${executable({ platform })} run`;
};

const importInstallScript = ({ platform = DEFAULT_PLATFORM }) => {
  switch (platform) {
    case LINUX_PLATFORM:
      return linuxInstall;
    case MACOS_PLATFORM:
      return osxInstall;
    case WINDOWS_PLATFORM:
      return windowsInstall;
    default:
      return '';
  }
};

export const platformArchitectures = ({ platform }) => {
  return DOWNLOAD_LOCATIONS[platform].map(({ arch }) => arch);
};

export const installScript = ({ platform, architecture }) => {
  const downloadLocation = DOWNLOAD_LOCATIONS[platform].find(({ arch }) => arch === architecture)
    .url;

  return importInstallScript({ platform })
    .replace(
      // eslint-disable-next-line no-template-curly-in-string
      '${GITLAB_CI_RUNNER_DOWNLOAD_LOCATION}',
      downloadLocation,
    )
    .trim();
};
