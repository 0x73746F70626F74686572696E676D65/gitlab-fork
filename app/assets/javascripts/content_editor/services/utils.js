export const hasSelection = (tiptapEditor) => {
  const { from, to } = tiptapEditor.state.selection;

  return from < to;
};

export const clamp = (n, min, max) => Math.max(Math.min(n, max), min);
