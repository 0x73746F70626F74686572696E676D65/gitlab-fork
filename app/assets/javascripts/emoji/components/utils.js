import { chunk, memoize } from 'lodash';
import { initEmojiMap, getEmojiCategoryMap } from '~/emoji';

export const generateCategoryHeight = (emojisLength) => emojisLength * 34 + 29;

export const getEmojiCategories = memoize(async () => {
  await initEmojiMap();

  const categories = await getEmojiCategoryMap();
  let top = 0;

  return Object.freeze(
    Object.keys(categories).reduce((acc, category) => {
      const emojis = chunk(categories[category], 9);
      const height = generateCategoryHeight(emojis.length);
      const newAcc = {
        ...acc,
        [category]: { emojis, height, top },
      };
      top += height;

      return newAcc;
    }, {}),
  );
});
