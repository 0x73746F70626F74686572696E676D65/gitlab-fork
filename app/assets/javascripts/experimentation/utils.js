// This file only applies to use of experiments through https://gitlab.com/gitlab-org/gitlab-experiment
import { get } from 'lodash';
import { DEFAULT_VARIANT, CANDIDATE_VARIANT, TRACKING_CONTEXT_SCHEMA } from './constants';

function getExperimentsData() {
  // Pull from deprecated window.gon.experiment
  const experimentsFromGon = get(window, ['gon', 'experiment'], {});
  // Pull from preferred window.gl.experiments
  const experimentsFromGl = get(window, ['gl', 'experiments'], {});

  return { ...experimentsFromGon, ...experimentsFromGl };
}

function convertExperimentDataToExperimentContext(experimentData) {
  return { schema: TRACKING_CONTEXT_SCHEMA, data: experimentData };
}

export function getExperimentData(experimentName) {
  return getExperimentsData()[experimentName];
}

export function getAllExperimentContexts() {
  return Object.values(getExperimentsData()).map(convertExperimentDataToExperimentContext);
}

export function isExperimentVariant(experimentName, variantName) {
  return getExperimentData(experimentName)?.variant === variantName;
}

export function getExperimentVariant(experimentName) {
  return getExperimentData(experimentName)?.variant || DEFAULT_VARIANT;
}

export function experiment(experimentName, { use, control, candidate, ...variants }) {
  const variant = getExperimentVariant(experimentName);

  switch (variant) {
    case DEFAULT_VARIANT:
      return (use || control).call();
    case CANDIDATE_VARIANT:
      return (variants.try || candidate).call();
    default:
      return variants[variant].call();
  }
}
