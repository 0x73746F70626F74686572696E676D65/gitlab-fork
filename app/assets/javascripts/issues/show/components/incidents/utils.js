import { createAlert } from '~/flash';
import { s__ } from '~/locale';

export const displayAndLogError = (error) =>
  createAlert({
    message: s__('Incident|Something went wrong while fetching incident timeline events.'),
    captureError: true,
    error,
  });

const EVENT_ICONS = {
  comment: 'comment',
  default: 'comment',
};

export const getEventIcon = (actionName) => {
  return EVENT_ICONS[actionName] ?? EVENT_ICONS.default;
};
