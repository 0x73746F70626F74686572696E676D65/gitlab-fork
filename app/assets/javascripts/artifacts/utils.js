import { numberToHumanSize } from '~/lib/utils/number_utils';
import { ARCHIVE_FILE_TYPE } from './constants';

export const totalArtifactsSizeForJob = (job) =>
  numberToHumanSize(
    job.artifacts.nodes
      .map((artifact) => artifact.size)
      .reduce((total, artifact) => total + artifact, 0),
  );

export const mapArchivesToJobNodes = (jobNode) => {
  return {
    archive: {
      ...jobNode.artifacts.nodes.find((artifact) => artifact.fileType === ARCHIVE_FILE_TYPE),
    },
    ...jobNode,
  };
};
