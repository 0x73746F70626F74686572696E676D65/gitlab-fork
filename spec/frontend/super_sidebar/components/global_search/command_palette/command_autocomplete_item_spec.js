import { shallowMount } from '@vue/test-utils';
import CommandAutocompleteItem from '~/super_sidebar/components/global_search/command_palette/command_autocomplete_item.vue';
import { linksReducer } from '~/super_sidebar/components/global_search/command_palette/utils';
import { LINKS } from './mock_data';

describe('CommandAutocompleteItem', () => {
  let wrapper;

  const createComponent = () => {
    wrapper = shallowMount(CommandAutocompleteItem, {
      propsData: {
        command: LINKS.reduce(linksReducer, [])[1],
        searchQuery: 'root',
      },
    });
  };

  beforeEach(() => {
    createComponent();
  });

  it('should render user item', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
