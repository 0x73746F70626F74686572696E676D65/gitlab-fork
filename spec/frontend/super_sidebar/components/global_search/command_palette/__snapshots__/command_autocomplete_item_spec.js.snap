// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandAutocompleteItem should render user item 1`] = `
<div
  class="gl-display-flex gl-align-items-center"
>
  <gl-icon-stub
    class="gl-mr-3"
    name="users"
    size="16"
  />
   
  <span
    class="gl-text-gray-900"
  >
    Manage &gt; Activity
  </span>
</div>
`;
